packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -79,100 +74,103 @@ export const askEnvironmentVariableCarryOut = async (
-    if (envVar === 'carry') return;
-    if (envVar === 'update') await envVarSelectFunction();
-  };
-
-  const envVarSelectFunction = async (): Promise<void> => {
-    const abortKey = uuid.v4();
-    const functionNameQuestion: inquirer.ListQuestion = {
-      type: 'list',
-      name: 'functionName',
-      message: 'Select the Lambda function you want to update values',
-      choices: functionNames
-        .map(name => ({
-          name,
-          value: name,
-        }))
-        .concat({
-          name: "I'm done",
-          value: abortKey,
-        }),
-    };
-    const { functionName } = await inquirer.prompt(functionNameQuestion);
-    if (functionName === abortKey) return;
-    await envVarSelectKey(functionName);
-  };
-
-  const envVarSelectKey = async (functionName: string): Promise<void> => {
-    const envVars = getStoredEnvironmentVariables(functionName, fromEnvName);
-    const abortKey = uuid.v4();
-    const keyNameQuestion: inquirer.ListQuestion = {
-      type: 'list',
-      name: 'keyName',
-      message: 'Which function\'s environment variables do you want to edit?',
-      choices: Object.keys(envVars)
-        .map(name => ({
-          name,
-          value: name,
-        }))
-        .concat({
-          name: "I'm done",
-          value: abortKey,
-        }),
-    };
-    const { keyName } = await inquirer.prompt(keyNameQuestion);
-    if (keyName === abortKey) {
-      await envVarSelectFunction();
-      return;
-    }
-    await envVarUpdateValue(functionName, keyName);
-  };
-
-  const envVarUpdateValue = async (functionName: string, keyName: string): Promise<void> => {
-    const envVars = getStoredEnvironmentVariables(functionName, fromEnvName);
-    const newValueQuestion: inquirer.InputQuestion = {
-      type: 'input',
-      name: 'newValue',
-      message: 'Enter the environment variable value:',
-      default: envVars[keyName],
-      validate: input => {
-        if (input.length >= 2048) {
-          return 'The value must be 2048 characters or less';
-        }
-        return true;
-      },
-    };
-    const { newValue } = await inquirer.prompt(newValueQuestion);
-    getEnvParamManager().getResourceParamManager(categoryName, functionName).setParam(_.camelCase(keyName), newValue);
-    await envVarSelectKey(functionName);
-  };
-
-  await envVarQuestion();
+  // eslint-disable-next-line spellcheck/spell-checker
+  if (context.parameters?.options?.quickstart) return;
+  if (yesFlagSet) return;
+
+  await askEnvVarCarryOrUpdateQuestion(functionNames, fromEnvName);
+};
+
+const askEnvVarCarryOrUpdateQuestion = async (functionNames: string[], fromEnvName: string): Promise<void> => {
+  const choices = [
+    {
+      value: 'carry',
+      name: 'Carry over existing environment variables to this new environment',
+    },
+    {
+      value: 'update',
+      name: 'Update environment variables now',
+    },
+  ];
+  const envVarOperation = await prompter.pick(
```

Comment:
In the previous version, the prompt is not executed if the yes flag is set, the amplify-prompts might achieve the same thing by passing a default value (not sure about this, if not we should not prompt on --yes flag and use 'carry' value)

Reply:
If we specify 'carry' to be the initial value for the pick prompt, it will be automatically selected if the --yes flag is present

Reply:
I actually ran into this while testing. If you look at the test that checks to make sure we aren't prompting, I ran into an infinite loop when not prompting. Is 'carry' default the preferred behavior?

Reply:
I will prefer the default in the prompt, otherwise it will become a bug one day if we decide to remove that check for some reason